src/constructs/core/parameters/ec2.ts

```diff
@@ -1,23 +1,26 @@
+import { AppIdentity } from "../identity";
 import type { GuStack } from "../stack";
-import type { GuNoTypeParameterProps } from "./base";
+import type { GuNoTypeParameterPropsWithAppIdentity } from "./base";
 import { GuParameter } from "./base";
 
 export class GuInstanceTypeParameter extends GuParameter {
-  constructor(scope: GuStack, id: string = "InstanceType", props?: GuNoTypeParameterProps) {
-    super(scope, id, {
+  // eslint-disable-next-line custom-rules/valid-constructors -- TODO be better
+  constructor(scope: GuStack, props: GuNoTypeParameterPropsWithAppIdentity) {
+    super(scope, AppIdentity.suffixText(props, "InstanceType"), {
       type: "String",
-      description: "EC2 Instance Type",
+      description: `EC2 Instance Type for the app ${props.app}`,
```

Comment:
For the app "janus", the ID/name of this parameter becomes "InstanceTypeJanus". Hopefully the description is helpful?